src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +225,296 @@ def execute(self, reqStatus):
+            self.logger.debug(msg, pformat(wflow))
+            total_num_requests += 1
+            if self._checkClean(wflow):
+                clean_num_requests += 1
+
+        # Report the counters:
+        for pline in self.cleanuplines:
+            msg = "%s: Workflows cleaned by pipeline: %s: %d"
+            self.logger.info(msg, self.currThreadIdent, pline.name, self.wfCounters['cleaned'][pline.name])
+        archived_num_requests = self.wfCounters['archived']
+        msg = "%s: Workflows archived: %d"
+        self.logger.info(msg, self.currThreadIdent, self.wfCounters['archived'])
+        return clean_num_requests, total_num_requests, archived_num_requests
+
+    def _dispatchWflow(self, wflow, *args, **kwargs):
+        """
+        A function intended to dispatch a workflow (e.g based on its status)
+        through one or more functional pipelines in case there is some more
+        complicated logic involved in the order we execute them but not just
+        a sequentially
+        """
+        # NOTE: The following dispatch logic is a subject to be changed at any time
+
+        if wflow['RequestType'] == 'Resubmission':
+            pass
+
+        # Clean:
+        elif wflow['RequestStatus'] in ['rejected', 'aborted-completed']:
+            # NOTE: We do not check the ParentageResolved flag for these
+            #       workflows, but we do need to clean output data placement
+            #       rules from the agents for them
+            for pline in self.agentlines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg, pline.name, str(ex))
+                    break
+                if wflow['CleanupStatus'][pline.name]:
+                    self.wfCounters['cleaned'][pline.name] += 1
+        elif wflow['RequestStatus'] == 'announced' and wflow['ParentageResolved']:
+            for pline in self.cleanuplines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg, pline.name, str(ex))
+                    break
+                if wflow['CleanupStatus'][pline.name]:
+                    self.wfCounters['cleaned'][pline.name] += 1
+
+        # Archive:
+        try:
+            self.plineArchive.run(wflow)
+            self.wfCounters['archived'] += 1
+        except MSRuleCleanerArchival as ex:
+            msg = "%s: Archival Error: %s. "
+            msg += "\nWill retry again in the next cycle."
+            self.logger.error(msg, wflow['PlineMarkers'][-1], ex.myMessage)
```

Comment:
I tried this and the logs become really blurred. It adds alot of extra (multi-lined) info comming from the WMCore Exception, including fetched job error codes and so on, which are totally not relevant to our case, and it will pop out in every workflow for which archival is skipped. it makes the whole log pretty much unreadable. See the difference between [1] and [2].

[1]
```
2020-11-04 08:38:49,651:DEBUG:MSRuleCleaner:cleanRucioRules(): plineAgentCont: 'TransferStatus' is 'pending' or 'TransferInfo' is missing. Skip deleting any rules for: tivanov_TC_LHE_PFN_HG2011_Val_201029_112700_5697 
2020-11-04 08:38:49,682:DEBUG:MSRuleCleaner:cleanRucioRules(): plineAgentBlock: 'TransferStatus' is 'pending' or 'TransferInfo' is missing. Skip deleting any rules for: tivanov_TC_LHE_PFN_HG2011_Val_201029_112700_5697 
2020-11-04 08:38:49,703:ERROR:MSRuleCleaner:_dispatchWflow(): plineArchive: Archival Error: <@========== WMException Start ==========@>
Exception Class: MSRuleCleanerArchival
Message: MSRuleCleanerArchival: Not properly cleaned workflow: tivanov_TC_LHE_PFN_HG2011_Val_201029_112700_5697
	ModuleName : WMCore.MicroService.Unified.MSRuleCleaner
	MethodName : __init__
	ClassInstance : None
	FileName : /afs/cern.ch/user/t/tivanov/work/private/WMCoreDev.d/WMCore/src/python/WMCore/MicroService/Unified/MSRuleCleaner.py
	ClassName : None
	LineNumber : 41
	ErrorNr : 0

Traceback: 

<@---------- WMException End ----------@>. 
 ```

[2]
```
2020-11-04 08:40:36,497:DEBUG:MSRuleCleaner:cleanRucioRules(): plineAgentBlock: 'TransferStatus' is 'pending' or 'TransferInfo' is missing. Skip deleting any rules for: tivanov_TC_LHE_PFN_HG2011_Val_201029_112700_5697 
2020-11-04 08:40:36,518:ERROR:MSRuleCleaner:_dispatchWflow(): plineArchive: Archival Error: MSRuleCleanerArchival: Not properly cleaned workflow: tivanov_TC_LHE_PFN_HG2011_Val_201029_112700_5697. 
Will retry again in the next cycle.
```

Reply:
I see. How about calling this method then?
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMException.py#L161

Reply:
It does the job. Thanks!